docs/content/Examples-Tutorials-Recipes/Building-context-aware-pre-aggregations.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Building Context-Aware Pre-aggregations
+permalink: /recipes/building-context-aware-pre-aggregations
+category: Examples & Tutorials
+subCategory: Query acceleration
+menuOrder: 6
+---
+
+## Use case
+
+Using different environments to test a product is good practice. Let's imagine
+we have three environments: testing, staging, and production. We want to use
+multiple database schemas depending on the current environment. In addition, we
+want to use pre-aggregation for each schema. So, in the recipe below, we'll
+learn how dynamically select database schema and create a pre-aggregation with
+the scheduled update.
+
+## Data schema
+
+To select the database schema, we will use the
+[`COMPILE_CONTEXT`](https://cube.dev/docs/schema/reference/cube#context-variables-compile-context)
+global variable. Let's pass into it an `env` variable from
+[`securityContext`](https://cube.dev/docs/security/context) with our environment
+value.
+
+```javascript
+const {
+  securityContext: { env },
+} = COMPILE_CONTEXT;
+
+cube(`Products`, {
+  sql: `SELECT * FROM ${env}.Orders`,
+  ...
+}
+```
+
+To significantly reduce response time, we will define the pre-aggregations.
+We'll also specify the
+[`refreshKey`](https://cube.dev/docs/schema/reference/pre-aggregations#parameters-refresh-key)
+option for keeping pre-aggregations up to date:
+
+```javascript
+preAggregations: {
+    amountByClientName: {
+      measures: [Products.amount],
+      dimensions: [Products.clientName],
+      timeDimension: Products.createdAt,
+      granularity: `day`,
+      refreshKey: {
+        every: `1 minute`,
+      },
+    },
+  }
+```
+
+## Configuration
+
+We'll use Cube Store as pre-aggregations storage with
+[Refresh Worker](https://cube.dev/docs/deployment/production-checklist#set-up-refresh-worker),
+updating our pre-aggregations in the background. The problem is Refresh Worker
+does not have a `securityContext` when he started. Because of this, he will not
+be able to figure out which scheme to use to build a pre-aggregation. To fix
+this, we can use the
+[`scheduledRefreshContexts`](https://cube.dev/docs/config#scheduled-refresh-contexts)
+function, which allows to generate the security contexts:
+
+```javascript
+module.exports = {
+  // Provides distinct identifiers for each tenant which are used as caching keys
+  contextToAppId: ({ securityContext }) => `CUBEJS_APP_${securityContext.env}`,
+
+  // Defines contexts for scheduled pre-aggregation update
+  scheduledRefreshContexts: async () => [
+    {
+      securityContext: {
+        env: 'testing',
+      },
+    },
+    {
+      securityContext: {
+        env: 'staging',
+      },
+    },
+    {
+      securityContext: {
+        env: 'production',
+      },
+    }
+  ]
+};
+```
+
+## Query
+
+Now we can send the query to get the client name and amount for `orders` table
+from multiple schemas using
+[JWT](https://cube.dev/docs/security#generating-json-web-tokens-jwt):
```

Comment:
Might be nice to show the example JWT for the query as well

Reply:
Make sense. Added JWT :)